swift/ql/src/queries/Summary/SummaryStats.ql

```diff
@@ -0,0 +1,25 @@
+/**
+ * @name Summary statistics
+ * @description A table of summary statistics about a database.
+ * @kind table
+ * @id swift/summary/summary-statistics
+ * @tags summary
+ */
+
+import swift
+import codeql.swift.dataflow.FlowSources
+import codeql.swift.security.SensitiveExprs
+
+predicate statistic(string what, int value) {
+  what = "Files" and value = count(File f)
+  or
+  what = "Expressions" and value = count(Expr e | e.getFile().getName() != "")
```

Comment:
For C++ we have:
```
class UnknownLocation extends Location {
  UnknownLocation() { this.getFile().getAbsolutePath() = "" }
}
```
Doesn't that exist for Swift?

Reply:
Indeed, Swift also has an `UnknownLocation` class: https://github.com/github/codeql/blob/30fc6acb19b5ddf8a23dd667d41b8750931bc643/swift/ql/lib/codeql/swift/elements/UnknownLocation.qll#L2.

Reply:
If it does, I should use it.  If it doesn't, I should add it...

Reply:
Thanks @MathiasVP !